autobot/api.py

```diff
@@ -9,76 +9,65 @@
 """Autobot API."""
 
 import yaml
+import copy
 
 from autobot.config_loader import Config
 from autobot.github import GitHubAPI
 
-
 class BotAPI:
-    """Generates report."""
+    """Bot's API."""
 
-    @classmethod
-    def __init__(cls, config: Config):
+    def __init__(self, config: Config):
         """Bot initialization."""
-        cls.config = config.config
-        cls.report = GitHubAPI(
-            cls.config["AUTOBOT_OWNER"], cls.config["AUTOBOT_GH_TOKEN"]
-        ).report(cls.load_repositories())
+        self.config = config
+        self.report = GitHubAPI(
+            self.config["AUTOBOT_OWNER"],
+            self.config["AUTOBOT_GH_TOKEN"],
+            self.config.repositories
+        ).report()
 
     @classmethod
-    def generate_report(cls, maintainer: str) -> dict:
-        """Generates a report for a maintainer."""
-        res = cls.report[0]["repos"]
-        print(res)
-        print(
-            "---------------------------------------------------------------------------------"
-        )
-        return res
+    def md_report(cls, report):
+        """Returns the report in markdown format."""
+        md_report = ""
```

Comment:
Just a small memory optimization here, instead of keeping the result in a string and concatenate it with `+=`, it's better to build a list of strings for each line and join them at the end. Here's how this might look like:

```python
lines = []
lines.append('# Some header')
for repo in repositories:
    lines.append(f'### {repo.name}')
    for action in repo.actions:
        lines.append(f'  - {action}')

# join all the lines now
result = '\n'.join(lines)
```